code/modules/mob/living/carbon/human/species_types/jellypeople.dm

```diff
@@ -479,88 +479,192 @@
+	else if(istype(target, /datum/mind))
+		var/datum/mind/mind_target = target
+		if(mind_transfer)
+			UnregisterSignal(mind_target, COMSIG_MIND_TRANSFER_TO)
+		if(mind_target.current)
+			UnregisterSignal(mind_target.current, list(COMSIG_MOB_LOGIN, COMSIG_MOB_DEATH))
+
+/datum/species/oozeling/stargazer/proc/validate_mind(datum/mind/mind, silent = FALSE)
+	. = TRUE
+	if(QDELETED(mind) || QDELETED(mind.current) || mind.current.stat == DEAD)
+		return FALSE
+	if(istype(mind.current.get_item_by_slot(ITEM_SLOT_HEAD), /obj/item/clothing/head/foilhat))
+		if(!silent)
+			var/mob/living/carbon/human/owner = slimelink_owner.resolve()
+			if(owner)
+				to_chat(mind.current, "<span class='danger'>[owner.real_name]'s no-good syndicate mind-slime is blocked by your protective headgear!</span>")
+		return FALSE
+	if(HAS_TRAIT_NOT_FROM(mind.current, TRAIT_MINDSHIELD, "nanites")) //mindshield implant, no dice
+		return FALSE
+
+/datum/species/oozeling/stargazer/proc/link_death(mob/living/source_mob)
+	SIGNAL_HANDLER
+	if(!source_mob.mind || !(source_mob.mind in linked_minds))
+		return
+	var/datum/weakref/mind_ref = GLOB.slime_linked_with[source_mob.mind]
+	var/datum/species/oozeling/stargazer/stargazer = mind_ref?.resolve()
+	var/mob/living/carbon/human/link_owner = stargazer?.slimelink_owner?.resolve()
+	if(!link_owner || !istype(link_owner))
+		return
+	to_chat("<span class='danger'>As you die, you feel your link to [link_owner.real_name] fizzle out!</span>")
+	unlink_mind(source_mob.mind)
 
 /datum/species/oozeling/stargazer/on_species_gain(mob/living/carbon/C, datum/species/old_species)
 	..()
 	project_thought = new(src)
 	project_thought.Grant(C)
 	link_minds = new(src)
 	link_minds.Grant(C)
+	linked_speech = new(src)
+	linked_speech.Grant(C)
 	slimelink_owner = WEAKREF(C)
-	link_mob(C)
 
-/datum/species/oozeling/stargazer/proc/link_mob(mob/living/M)
-	if(QDELETED(M) || M.stat == DEAD)
-		return FALSE
-	if(HAS_TRAIT(M, TRAIT_MINDSHIELD)) //mindshield implant, no dice
+/datum/species/oozeling/stargazer/proc/link_mind(datum/mind/mind)
+	if(!validate_mind(mind))
 		return FALSE
 	var/mob/living/carbon/human/owner = slimelink_owner.resolve()
-	if(istype(M.get_item_by_slot(ITEM_SLOT_HEAD), /obj/item/clothing/head/foilhat))
-		if(owner)
-			to_chat(M, "<span class='danger'>[owner.real_name]'s no-good syndicate mind-slime is blocked by your protective headgear!</span>")
-
-		return FALSE
-	if(M in linked_mobs)
+	if(mind in linked_minds)
 		return FALSE
 	if(!owner)
 		return FALSE
-	linked_mobs.Add(M)
-	to_chat(M, "<span class='notice'>You are now connected to [owner.real_name]'s Slime Link.</span>")
+	linked_minds.Add(mind)
 	var/datum/action/innate/linked_speech/action = new(src)
 	linked_actions.Add(action)
-	action.Grant(M)
+	action.Grant(mind.current)
+	to_chat(mind.current, "<span class='notice'>You are now connected to [owner.real_name]'s Slime Link.</span>")
+	GLOB.slime_linked_with[mind] = WEAKREF(src)
```

Comment:
I feel like you made a mistake here. Did you intend to `+=` here? I feel like only one person will be linked to a mind

Reply:
```
/// A global list of what mind is linked with what stargazer.
/// [mind] = /datum/weakref -> /datum/species/oozeling/stargazer
GLOBAL_LIST_EMPTY(slime_linked_with)
```

Reply:
it's not mind weak ref.

Reply:
correct...?

the key is a mind, the value is a weakref to a stargazer species datum.

Reply:
you're answering to outdated comment.